specification/common-types/resource-management/v3/managedidentity.json

```diff
@@ -108,6 +108,47 @@
       "required": [
         "type"
       ]
+    },
+
+    "SystemAssignedOrUserAssignedServiceIdentityType": {
+      "description": "Type of managed service identity (either system assigned, or none).",
+      "enum": [
+        "None",
+        "SystemAssigned",
+        "UserAssigned"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "SystemAssignedOrUserAssignedServiceIdentityType",
+        "modelAsString": true
+      }
+    },
+    "SystemAssignedOrUserAssignedServiceIdentity": {
+      "description": "Type of managed service identity (system assigned or user assigned identities - but not system assigned and user assigned identities)",
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "readOnly": true,
+          "format": "uuid",
+          "type": "string",
+          "description": "The service principal ID of the system assigned identity. This property will only be provided for a system assigned identity."
+        },
+        "tenantId": {
+          "readOnly": true,
+          "format": "uuid",
+          "type": "string",
+          "description": "The tenant ID of the system assigned identity. This property will only be provided for a system assigned identity."
+        },
+        "type": {
+          "$ref": "#/definitions/SystemAssignedOrUserAssignedServiceIdentityType"
+        },
+        "userAssignedIdentities": {
+          "$ref": "#/definitions/UserAssignedIdentities"
+        }
+      },
+      "required": [
+        "type"
+      ]
```

Comment:
just put the definition in your services folders.  Best practice is to support both

Reply:
@pilor we've seen this in other Services too, where the Service is currently using the incorrect Identity type since there isn't a common Identity type for this scenario - so whilst we can do that, having an Identity type duplicated in each service seems likely to cause further behavioural discrepancies?

Reply:
Presence in commonTypes is going to imply blessing of the pattern.  Its not the suggested pattern and the identity docs specifically call out downsides with this approach.  Multiple identities should be supported and they can be a mix of both types.  If that changes in the future (i.e. we start trying to get people off systemAssigned) then it would make sense for it to be in commonTypes.

I appreciate the intent here but its going to be a review headache when services start pointing to this definition as proof that the pattern is "correct".